Brixcer_Box/src/LedControl.cpp

```diff
@@ -2,18 +2,38 @@
 #include "LedControl.h"
 
 
-LedControl::LedControl(int ledPin) {
-    this->ledPin = ledPin;
+Adafruit_NeoPixel LedControl::strip;
 
-    pinMode(this->ledPin, OUTPUT);
+LedControl::LedControl(int ringNum) {
+    this->ringNum = ringNum;
+}
+
+void LedControl::InitLedControl(int count, int pin) {
+    // Parameter 1 = number of pixels in strip
+    // Parameter 2 = pin number (most are valid)
+    // Parameter 3 = pixel type flags, add together as needed:
+    //   NEO_KHZ800  800 KHz bitstream (most NeoPixel products w/WS2812 LEDs)
+    //   NEO_KHZ400  400 KHz (classic 'v1' (not v2) FLORA pixels, WS2811 drivers)
+    //   NEO_GRB     Pixels are wired for GRB bitstream (most NeoPixel products)
+    //   NEO_RGB     Pixels are wired for RGB bitstream (v1 FLORA pixels, not v2)
+    LedControl::strip = Adafruit_NeoPixel(LEDS_PER_RING * count, pin, NEO_GRB + NEO_KHZ800);
+
+    strip.begin();
+    strip.setBrightness(1);
+    strip.show();
 }
 
 void LedControl::setVolume(int volume) {
-    this->volume = volume;
+    int volumeStartLed = this->ringNum * LEDS_PER_RING + APPLICATION_LEDS;
+    int volumeLedRange = LEDS_PER_RING - APPLICATION_LEDS;
```

Comment:
Thx for making me feel stupid with my hacked together prototype of it. Probably should have spent more time on that.

Reply:
What are friends for?